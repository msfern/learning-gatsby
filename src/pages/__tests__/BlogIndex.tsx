import React from "react";
import renderer from "react-test-renderer";
import * as Gatsby from "../../../__mocks__/gatsby";
import BlogIndex from "../blog";

beforeEach(() => {
  const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery");
  useStaticQuery.mockImplementationOnce(() => ({
    site: {
      siteMetadata: {
        title: "My Title",
      },
    },
    allMdx: {
      nodes: [
        {
          parent: {
            modifiedTime: "February 6, 2022",
          },
          frontmatter: {
            date: "February 1, 2022",
            title: "My first job abroad!",
          },
          id: "a7670ea3-30e8-578a-af1b-36362fbf28fe",
          slug: "job-abroad/",
        },
        {
          parent: {
            modifiedTime: "February 6, 2022",
          },
          frontmatter: {
            date: "December 25, 2021",
            title: "Mental Health",
          },
          id: "d56ad04a-8e60-56f7-9718-b51e489a47a7",
          slug: "mental-health/",
        },
        {
          parent: {
            modifiedTime: "February 6, 2022",
          },
          frontmatter: {
            date: "October 24, 2021",
            title: "My Guinea Pigs",
          },
          id: "6ffa82b7-b83b-5968-ae87-315e6b8b1fd1",
          slug: "my-guinea-pigs/",
        },
      ],
    },
  }));
});

describe("Blog", () => {
  it("renders correctly", () => {
    const data = {
      site: {
        siteMetadata: {
          title: "My Title",
        },
      },
      allMdx: {
        nodes: [
          {
            parent: {
              modifiedTime: "February 6, 2022",
            },
            frontmatter: {
              date: "February 1, 2022",
              title: "My first job abroad!",
            },
            id: "a7670ea3-30e8-578a-af1b-36362fbf28fe",
            slug: "job-abroad/",
          },
          {
            parent: {
              modifiedTime: "February 6, 2022",
            },
            frontmatter: {
              date: "December 25, 2021",
              title: "Mental Health",
            },
            id: "d56ad04a-8e60-56f7-9718-b51e489a47a7",
            slug: "mental-health/",
          },
          {
            parent: {
              modifiedTime: "February 6, 2022",
            },
            frontmatter: {
              date: "October 24, 2021",
              title: "My Guinea Pigs",
            },
            id: "6ffa82b7-b83b-5968-ae87-315e6b8b1fd1",
            slug: "my-guinea-pigs/",
          },
        ],
      },
    };

    const tree = renderer.create(<BlogIndex data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
