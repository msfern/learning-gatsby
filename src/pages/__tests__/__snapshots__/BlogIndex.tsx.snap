// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog renders correctly 1`] = `
<div
  className="container"
  data-testid="hero-title"
>
  <title>
    My Blog Posts
     | 
    My Title
  </title>
  <header
    className="siteTitle"
  >
    My Title
  </header>
  <nav>
    <ul
      className="navLinks"
    >
      <li
        className="navLinkItem"
      >
        <a
          className="navLinkText"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        className="navLinkItem"
      >
        <a
          className="navLinkText"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        className="navLinkItem"
      >
        <a
          className="navLinkText"
          href="/blog"
        >
          Blog
        </a>
      </li>
    </ul>
  </nav>
  <main>
    <h1
      className="heading"
    >
      My Blog Posts
    </h1>
    <article>
      <h2>
        <a
          href="/blog/job-abroad/"
        >
          My first job abroad!
        </a>
      </h2>
      <p>
        Posted: 
        February 1, 2022
      </p>
    </article>
    <article>
      <h2>
        <a
          href="/blog/mental-health/"
        >
          Mental Health
        </a>
      </h2>
      <p>
        Posted: 
        December 25, 2021
      </p>
    </article>
    <article>
      <h2>
        <a
          href="/blog/my-guinea-pigs/"
        >
          My Guinea Pigs
        </a>
      </h2>
      <p>
        Posted: 
        October 24, 2021
      </p>
    </article>
  </main>
</div>
`;
